feedback/feedback.py

```diff
@@ -102,141 +117,81 @@ def resource_string(cls, path):
-                              "material."),
-            'likert': _("Please rate your overall experience"),
-            'scale_text': [_("Excellent"),
-                           _("Good"),
-                           _("Average"),
-                           _("Fair"),
-                           _("Poor")],
-            'icon_set': 'num',
-            'placeholder': "Please take a moment to thoughtfully reflect."
-        }
-
-        prompt.update(self.prompts[index])
-        return prompt
-
     def student_view(self, context=None):
         """
         The primary view of the FeedbackXBlock, shown to students
         when viewing courses.
         """
-        # Figure out which prompt we show. We set self.prompt_choice to
-        # the index of the prompt. We set it if it is out of range (either
+        # Figure out which prompts we show. We set self.prompts_choice to
+        # the index of the prompts. We set it if it is out of range (either
         # uninitiailized, or incorrect due to changing list length). Then,
-        # we grab the prompt, prepopulated with defaults.
-        if self.prompt_choice < 0 or self.prompt_choice >= len(self.prompts):
-            self.prompt_choice = random.randint(0, len(self.prompts) - 1)
-        prompt = self.get_prompt()
+        # we grab the prompts, prepopulated with defaults.
+        propts_size = ASSESSMENTS_NUMBER
 
-        # Now, we render the FeedbackXBlock.
-        html = self.resource_string("static/html/feedback.html")
+        if self.prompts_choice < 0 or self.prompts_choice >= propts_size:
+            self.prompts_choice = random.randint(0, propts_size - 1)
 
-        # Staff see vote totals, so we have slightly different HTML here.
-        if self.vote_aggregate and self.is_staff():
-            scale_item = self.resource_string("static/html/staff_item.html")
-        else:
-            scale_item = self.resource_string("static/html/scale_item.html")
-        # The replace allows us to format the HTML nicely without getting
-        # extra whitespace
-        scale_item = scale_item.replace('\n', '')
-
-        # We have five Likert fields right now, but we'd like this to
-        # be dynamic
-        indexes = range(5)
-
-        # If the user voted before, we'd like to show that
-        active_vote = ["checked" if i == self.user_vote else ""
-                       for i in indexes]
+        indices = range(propts_size) if self.enable_zero_grade else range(1, propts_size)
+        active_vote = ["checked" if i == self.user_vote else "" for i in indices]
 
         # Confirm that we do have vote totals (this may be uninitialized
         # otherwise). This should probably go into __init__ or similar.
         self.init_vote_aggregate()
-        votes = self.vote_aggregate
+        votes = self.vote_aggregate if self.enable_zero_grade else self.vote_aggregate[1:]
+
+        icon_path_template = {
+            "inactive": "public/numbers/i{set}{i}.png",
```

Comment:
Do we miss here the formatting?

Reply:
No, we format this string below in the `get_url ` method